bigquery/scheduled_queries/calculate-daily-mat-time-metrics.sql

```diff
@@ -0,0 +1,362 @@
+      COUNTIF(using_mat_access_workaround)) AS trusts_using_mat_access_workaround,
+  IF
+    (date > CURRENT_DATE(),
+      NULL,
+      COUNTIF(has_published_so_far)) AS trusts_which_published_so_far,
+  IF
+    (date > CURRENT_DATE(),
+      NULL,
+      COUNTIF(has_published_in_the_last_year)) AS trusts_which_published_in_the_last_year,
+  IF
+    (date > CURRENT_DATE(),
+      NULL,
+      COUNTIF(has_published_in_the_last_quarter)) AS trusts_which_published_in_the_last_quarter,
+  IF
+    (date > CURRENT_DATE(),
+      NULL,
+      COUNTIF(had_live_vacancies)) AS trusts_which_had_vacancies_live,
+  IF
+    (date > CURRENT_DATE(),
+      NULL,
+      COUNTIF(has_published_multischool_vacancies_so_far)) AS trusts_which_published_multischool_vacancies_so_far,
+  IF
+    (date > CURRENT_DATE(),
+      NULL,
+      COUNTIF(has_published_multischool_vacancies_in_the_last_year)) AS trusts_which_published_multischool_vacancies_in_the_last_year,
+  IF
+    (date > CURRENT_DATE(),
+      NULL,
+      COUNTIF(has_published_multischool_vacancies_in_the_last_quarter)) AS trusts_which_published_multischool_vacancies_in_the_last_quarter,
+  IF
+    (date > CURRENT_DATE(),
+      NULL,
+      COUNTIF(had_live_multischool_vacancies)) AS trusts_which_had_multischool_vacancies_live,
+  IF
+    (date > CURRENT_DATE(),
+      NULL,
+      COUNTIF(has_published_trust_level_vacancies_so_far)) AS trusts_which_published_trust_level_vacancies_so_far,
+  IF
+    (date > CURRENT_DATE(),
+      NULL,
+      COUNTIF(has_published_trust_level_vacancies_in_the_last_year)) AS trusts_which_published_trust_level_vacancies_in_the_last_year,
+  IF
+    (date > CURRENT_DATE(),
+      NULL,
+      COUNTIF(has_published_trust_level_vacancies_in_the_last_quarter)) AS trusts_which_published_trust_level_vacancies_in_the_last_quarter,
+  IF
+    (date > CURRENT_DATE(),
+      NULL,
+      COUNTIF(had_live_trust_level_vacancies)) AS trusts_which_had_trust_level_vacancies_live
+  FROM (
+    SELECT
+      trusts.uid AS uid,
+      dates.date AS date,
+      trusts.size_bracket AS size_bracket,
+      trusts.using_mat_access_workaround AS using_mat_access_workaround,
+    IF
+      ((trusts.status != "Closed"
+          OR trusts.date_closed > dates.date) #if the trust is not currently closed or if it closed after the date we're calculating for
+        AND (trusts.status != "Proposed to open" #and if the trust is not currently proposed to open
+          OR trusts.date_opened <= dates.date #or if the trust opened before the date we're calculating for
+          )
+        AND trusts.date_created <= dates.date,
+        #and if the trust was first listed on GIAS before or on the date we're calculating for
+        TRUE,
+        FALSE) AS in_scope,
+    IF
+      ((trusts.status = "Closed"
+          AND trusts.date_closed <= dates.date)
+        OR trusts.date_opened > dates.date,
+        FALSE,
+        #mark trusts which had closed or had not yet opened as not signed up
```

Comment:
If you mark trusts which are closed as not signed up, does that give a misleading impression of the proportion of trusts which we still want to sign up?

Reply:
(same q applies for trust which are closed marked as not having published)

Reply:
@david-mears-dfe in the case, we're calculating the trusts which *are* signed up as a proportion of the trusts in scope, both of which exclude closed trusts, so the figures we're producing aren't actually misleading I think. But good question!

Reply:
Let me see if i can change the comment to make this clearer

Reply:
is there a way to only need to exclude the closed trusts once rather than repeatedly?

Reply:
Not simply, unless you've got an idea? The complexity is that it's not as simple as just excluding closed trusts, because which trusts are closed changes for each date we're calculating for - I think we'd need to add a subquery in a WITH statement before this one that produced a table of dates and trusts that were open on those dates. But while that would be nicer to read, it would end up being executed multiple times and so actually be less efficient SQL I think. BigQuery imposes limits on the number of subquery executions you can do in any one query to encourage you to be efficient, hence why I'm trying to keep this kind of thing down.

Reply:
@david-mears-dfe I found a way :-) Please could you re-review?